models/ssh_key.go

```diff
@@ -557,53 +557,53 @@ func RewriteAllPublicKeys() error {
 	sshOpLocker.Lock()
 	defer sshOpLocker.Unlock()
 
-	fpath := filepath.Join(setting.SSH.RootPath, "authorized_keys")
-	tmpPath := fpath + ".tmp"
-	f, err := os.OpenFile(tmpPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+	fPath := filepath.Join(setting.SSH.RootPath, "authorized_keys")
+	tmpPath := fPath + ".tmp"
+	t, err := os.OpenFile(tmpPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
 	if err != nil {
 		return err
 	}
 	defer func() {
-		f.Close()
+		t.Close()
 		os.Remove(tmpPath)
 	}()
 
+	if com.IsExist(fPath) && setting.SSH.AuthorizedKeysBackup {
```

Comment:
It would be better: `if setting.SSH.AuthorizedKeysBackup && com.IsExist(fPath) {` to not check for directory existence if backup is disabled anyway, disk i/o operations are not cheap especially on rpi